### PR TITLE
E2e eslint

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The e2e tests have no lint rules.  This meant that errors wer egoing unnoticed.  E.g. one check was always passing because a promise was being interpreted as `true`.

# Changes
- Import the `.eslint` rules from `frontend/svelte` with the following changes:
  - Removed svelte rules, as they are not needed here.
  - Added the wdio eslint rules - although they haven't changed any eslint results so far.
  - Allowed argument starting with _ to be unused.
- fixed lint errors
- Added an empty .prettierrc.js, as otherwise the formatter picks up the system defaults.  I like tabs. :-)

Note: I thought that we already allowed arguments starting with _ to be unused.  We definitely discussed and agreed but the code may never have landed.

Note: The strict bool is clearly useful in some cases, however it leaves us with no nice way of expressing defaults, as far as I know.  E.g. `process.env.WHATEVER || "foo"` gets a lint error.   Suggestions are welcome.

# Tests
There SHOULD be no functional changes.  Any errors SHOULD be picked up by CI.

### CODE REVIEW

File: e2e-tests/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+module.exports = {
+  root: true,
+  parser: "@typescript-eslint/parser",
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    // TODO: Enable once we fix issues from above
+    // "plugin:@typescript-eslint/recommended-requiring-type-checking",
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply e2e-tests/.eslintrc.js:9]: I had the same for the svelte project. Definitely in another PR.

File: e2e-tests/specs/redirect-to-legacy.e2e.ts [hunk 2]
```diff
@@ -18,8 +18,11 @@ enum RedirectToLegacy {
   svelte,
   both,
 }
+
 const REDIRECT_TO_LEGACY: RedirectToLegacy =
-  process.env.REDIRECT_TO_LEGACY || RedirectToLegacy.prod;
+  RedirectToLegacy[process.env.REDIRECT_TO_LEGACY] !== undefined
+    ? RedirectToLegacy[process.env.REDIRECT_TO_LEGACY]
+    : RedirectToLegacy.prod;
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply e2e-tests/specs/redirect-to-legacy.e2e.ts:25]: Maybe create a file `constants.ts` where we import `REDIRECT_TO_LEGACY`, then cast there?

We do that in svelte:
```
const deployEnvironment: string = String(process.env.DEPLOY_ENV);
```

File: e2e-tests/tsconfig.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
       "@wdio/mocha-framework",
       "expect-webdriverio"
     ],
+    "strictNullChecks": true,
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply e2e-tests/tsconfig.json:9]: Yes, it is needed for the lint rule: `@typescript-eslint/strict-boolean-expressions`

File: e2e-tests/wdio.conf.ts [hunk 4]
```diff
@@ -1,17 +1,21 @@
 import { existsSync, mkdirSync } from "fs";
 
+const NNS_DAPP_URL: string = (process.env.NNS_DAPP_URL as boolean)
```
Comment (lmuntanerDfinity) [inline e2e-tests/wdio.conf.ts:3]: ditto on moving this to a `constants.ts` and casting there.

**Review by lmuntanerDfinity — APPROVED** [review]

File: e2e-tests/test.js [hunk 5]
```diff
@@ -33,7 +34,7 @@ let replica_host;
 try {
   const dfx_json = JSON.parse(fs.readFileSync(DFX_JSON_PATH, "utf8"));
   replica_host = dfx_json.networks.local.bind;
-  if (!replica_host.startsWith("http://")) {
+  if (replica_host.startsWith("http://") !== true) {
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply e2e-tests/test.js:37]: Do we care about https?
```js
if (/^https?:\/\//.test(replica_host) === false) {
  // ...
}
```

[the remark this answers is not in the document]
Comment (bitdivine) [reply e2e-tests/test.js]: Evidently no-one has cared about it so far but we can remove the unpleasant surprise here when someone does want to use https.  Done.

**Review by mstrasinskis — APPROVED** [review]
LGTM

File: e2e-tests/wdio.conf.ts (hunk 4 above)
Comment (bitdivine) [reply e2e-tests/wdio.conf.ts]: Done.

**Pull request merged**